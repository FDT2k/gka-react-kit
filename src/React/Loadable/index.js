import React from 'react';
import Loadable from 'react-loadable';
import Loading from '../Widgets/Loading'

export default  (loader)=>{
  const LoadableComponent =  Loadable({
    loader: loader,

    loading: (props)=>{
      if (props.error) {
        return <div>Error! {props.error.stack}<button onClick={ props.retry }>Retry</button></div>;
      } else if (props.pastDelay) {
        return (<Loading/>);
      } else {
        return null;
      }

    },
    delay: 500
  })


  return props=>(<LoadableComponent {...props}/>)
}
