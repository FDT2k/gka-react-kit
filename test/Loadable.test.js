// Link.react.test.js
import "@babel/polyfill"
import React from 'react';
import { Provider } from 'react-redux';
global.window = {}
import 'mock-local-storage'


import renderer from 'react-test-renderer';

class TestComponent extends React.Component {


  render() {
    return (
      <h1>test</h1>
    );
  }
}

import Loadable from '../src/React/Loadable'

test('Loadable is rendering', (done) => {
  return new Promise ((resolve,reject)=>{
    const C = Loadable(()=>import('./TestComponent'))
    console.log(C)
    const component = renderer.create(
        <C />
    );
    setTimeout(()=>{
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
      resolve();
    },1000)
  });
});
