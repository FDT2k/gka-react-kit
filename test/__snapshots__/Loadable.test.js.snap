// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loadable is rendering 1`] = `
<div>
  Error! 
  ReferenceError: React is not defined
    at Object.&lt;anonymous&gt; (/home/fabien/Documents/work/gka-react-redux-starter/test/TestComponent.js:1:1790)
    at Runtime._execModule (/home/fabien/Documents/work/gka-react-redux-starter/node_modules/jest-runtime/build/index.js:888:13)
    at Runtime._loadModule (/home/fabien/Documents/work/gka-react-redux-starter/node_modules/jest-runtime/build/index.js:577:12)
    at Runtime.requireModule (/home/fabien/Documents/work/gka-react-redux-starter/node_modules/jest-runtime/build/index.js:433:10)
    at Runtime.requireModuleOrMock (/home/fabien/Documents/work/gka-react-redux-starter/node_modules/jest-runtime/build/index.js:598:21)
    at /home/fabien/Documents/work/gka-react-redux-starter/test/Loadable.test.js:25:24
    at process._tickCallback (internal/process/next_tick.js:68:7)
  <button
    onClick={[Function]}
  >
    Retry
  </button>
</div>
`;
